signer/src/main.rs

```diff
@@ -78,15 +78,26 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     );
 
     // Load the configuration file and/or environment variables.
-    let settings = Settings::new(args.config)?;
+    let settings = Settings::new(args.config).unwrap_or_else(|error| {
+        tracing::error!(%error, "failed to construct the configuration");
+        std::process::exit(1);
+    });
```

Comment:
nit: we can use `inspect_err` + `?` so that we still get the usual output on stdout regardless of the logs
```suggestion
    let settings = Settings::new(args.config).inspect_err(|error| {
        tracing::error!(%error, "failed to construct the configuration");
    })?;
```

Reply:
We shouldn't need to use `process::exit()` here as `main()` returns a `Result<>` -- using the `?` coercion is enough.